ambry-store/src/main/java/com/github/ambry/store/BlobStoreCompactor.java

```diff
@@ -194,6 +197,24 @@ void compact(CompactionDetails details, byte[] bundleReadBuffer) throws IOExcept
     resumeCompaction(bundleReadBuffer);
   }
 
+  /**
+   * Filters not active {@link Container}s for compaction purpose. Not active status include DELETE_IN_PROGRESS and INACTIVE.
+   * @return the not active {@link Container}s' accountId & containerId pairs.
+   */
+  private void getNotActiveContainers() {
+    notActiveContainers.clear();
+    Set<Pair<Short, Short>> selectedContainerPairSet = new HashSet<>();
```

Comment:
nit: we don't need this new variable, you can just add the pair of account id and container id to the notActiveContainers set.

Reply:
+1

Reply:
Updated.